feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, List
+import logging
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(
+        self,
+        impute_with: Any = np.nan,
+        missing_values: str = "ignore",
+        drop_if_more_columns: bool = True,
+        add_if_less_columns: bool = True,
+        verbose: bool = False
+    ):
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError("missing_values takes only values 'raise' or 'ignore'.")
+
+        if not isinstance(drop_if_more_columns, bool):
+            raise ValueError("drop_if_more_columns takes only booleans True and False")
+
+        if not isinstance(add_if_less_columns, bool):
+            raise ValueError("add_if_less_columns takes only booleans True and False")
+
+        if not isinstance(verbose, bool):
+            raise ValueError("verbose takes only booleans True and False")
+
+        self.impute_with = impute_with
+        self.missing_values = missing_values
+        self.drop_if_more_columns = drop_if_more_columns
+        self.add_if_less_columns = add_if_less_columns
+        self.verbose = verbose
+
+    def _check_input(self, X: pd.DataFrame):
+        X = _is_dataframe(X)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        return X
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """Fit columns schema
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+        """
+        X = self._check_input(X)
+
+        self.variables_ = X.columns
+        return self
+
+    def _raise_errors_if_needed(
+        self,
+        _columns_to_drop: List[str],
+        _columns_to_add: List[str],
+        X: pd.DataFrame
+    ):
+
+        if not self.drop_if_more_columns and len(_columns_to_drop) > 0:
+            raise ValueError(
+                "drop_if_more_columns is set to False"
+                " and transform df contain more columns "
+                "than the df used in fit "
+                f"(found {set(X.columns)}, "
+                f"expected ({set(self.variables_)}))."
+            )
+
+        if not self.add_if_less_columns and len(_columns_to_add) > 0:
```

Comment:
same as per previous comment.

In my mind, if we allow the user to decide not to add the missing columns, then we should allow this functionality. But I may be understanding the logic wrongly. Please let me know your thoughts